test/e2e/router_compression_test.go

```diff
@@ -0,0 +1,261 @@
+	compressionPolicyNeedQuotes := operatorv1.HTTPCompressionPolicy{MimeTypes: mimeTypesNeedQuotes}
+
+	mimeTypesErrors := []operatorv1.CompressionMIMEType{"text/", "x- /value", "//"}
+	compressionPolicyErrors := operatorv1.HTTPCompressionPolicy{MimeTypes: mimeTypesErrors}
+
+	testParsing(t, "http-compression-1", compressionPolicyNormative, "error testing normal compression policy: %v")
+	testParsing(t, "http-compression-2", compressionPolicyEmpty, "error testing empty compression policy: %v")
+	testParsing(t, "http-compression-3", compressionPolicyNeedQuotes, "error testing compression policy that needs quotes: %v")
+
+	pc4, err := getPrivateController(t, "http-compression-4", dnsConfig.Spec.BaseDomain)
+	if err != nil {
+		t.Fatalf("error getting private controller: %v", err)
+	}
+	if err := testCompressionPolicy(t, "http-compression-4", compressionPolicyErrors); err == nil {
+		t.Errorf("compression policy with errors should have failed but didn't")
+	}
+	defer assertIngressControllerDeleted(t, kclient, pc4)
+}
+
+func testParsing(t *testing.T, name string, policy operatorv1.HTTPCompressionPolicy, errorMsg string) {
+	t.Helper()
+
+	pc, err := getPrivateController(t, name, dnsConfig.Spec.BaseDomain)
+	if err != nil {
+		t.Fatalf("error getting private controller: %v", err)
+	}
+	if err := testCompressionPolicy(t, name, policy); err != nil {
+		t.Errorf(fmt.Sprintf(errorMsg, err))
+	}
+	defer assertIngressControllerDeleted(t, kclient, pc)
+}
+
+func getPrivateController(t *testing.T, privateName string, privateDomain string) (*operatorv1.IngressController, error) {
+	t.Helper()
+
+	icName := types.NamespacedName{Namespace: operatorNamespace, Name: privateName}
+	domain := icName.Name + "." + privateDomain
+
+	ic := newPrivateController(icName, domain)
+	// Create a new private Ingress Controller (deletion handled by caller)
+	if err := kclient.Create(context.TODO(), ic); err != nil {
+		return ic, errors.New(fmt.Sprintf("error creating private ingresscontroller %s: %v", privateName, err))
+	}
+	return ic, nil
+}
+
+// testCompressionPolicy updates the given ingresscontroller's spec.httpCompressionPolicy with the given
+// compressionPolicy, and reports an error if it the compression policy was not accepted or not added to the router
+// deployment environment.
+func testCompressionPolicy(t *testing.T, name string, compressionPolicy operatorv1.HTTPCompressionPolicy) error {
+	t.Helper()
+	namespacedName := types.NamespacedName{Namespace: operatorNamespace, Name: name}
+	routerDeploymentNamespacedName := types.NamespacedName{Namespace: controller.DefaultOperandNamespace, Name: "router-" + name}
+
+	if err := wait.PollImmediate(10*time.Second, 2*time.Minute, func() (bool, error) {
+		ic, err := getIngressController(t, kclient, namespacedName, 5*time.Second)
+		if err != nil {
+			t.Logf("failed to get ingress controller: %v, retrying...", err)
+			return false, nil
+		}
+		ic.Spec.HTTPCompression = compressionPolicy
```

Comment:
```suggestion
		compressionPolicy.DeepCopyInto(&ic.Spec.HTTPCompression)
```
See #689 for why this sort of assignment is problematic.

Reply:
👍🏼